packages/grid/data-grid/src/tests/pagination.DataGrid.test.tsx

```diff
@@ -320,6 +322,55 @@ describe('<DataGrid /> - Pagination', () => {
         );
         expect(getColumnValues(0)).to.deep.equal(['7', '8']);
       });
+
+      it('should update the amount of rows rendered and call onPageSizeChange when changing the table height', async () => {
+        const onPageSizeChange = spy();
+
+        const TestCaseAutoPageSize = (props: { nbRows: number; height?: number }) => {
+          const data = useData(props.nbRows, 10);
+
+          return (
+            <div style={{ width: 300, height: props.height }}>
+              <DataGrid
+                columns={data.columns}
+                rows={data.rows}
+                autoPageSize
```

Comment:
It eases the calculation using explicit values.

```suggestion
                autoPageSize
                rowHeight={50}
                headerHeight={50}
```